bpf/aya-bpf/src/programs/probe.rs

```diff
@@ -3,19 +3,177 @@ use core::ffi::c_void;
 use crate::{bindings::pt_regs, BpfContext};
 
 pub struct ProbeContext {
-    pub regs: *mut pt_regs,
+    pub regs: Regs,
 }
 
 impl ProbeContext {
     pub fn new(ctx: *mut c_void) -> ProbeContext {
         ProbeContext {
-            regs: ctx as *mut pt_regs,
+            regs: Regs::from(ctx)
         }
     }
 }
 
 impl BpfContext for ProbeContext {
     fn as_ptr(&self) -> *mut c_void {
+        self.regs.as_raw_ptr()
+    }
+}
+
+pub struct Regs {
+    regs: *mut pt_regs,
+}
+
+impl From<*mut c_void> for Regs {
+    fn from(ctx: *mut c_void) -> Self {
+        Regs { regs: ctx as *mut pt_regs }
+    }
+}
+
+/// This struct allow to generate the macros we have in C as PT_REGS_*
+/// as simple method calls.
+impl Regs {
+    /// Utility to know if the context is valid
+    pub fn is_null(&self) -> bool {
+        self.regs.is_null()
+    }
+
+    /// Utility to get the raw ptr, useful to interact with the context directly.
+    pub fn as_raw_ptr(&self) -> *mut c_void {
         self.regs as *mut c_void
     }
+
+    /// Utility to get the underlying regs, useful for advanced users to access the
+    /// underlying binding without having to recast.
+    pub fn as_regs(&self) -> *mut pt_regs {
```

Comment:
I don't think we need this

Reply:
I made it crate only, I use it internally